src/peep/peep.c

```diff
@@ -8414,31 +8699,40 @@ static int guest_path_finding(rct_peep* peep)
 	}
 
 	uint8 edges = path_get_permitted_edges(mapElement);
+
+	if (edges == 0) {
```

Comment:
It's  minor optimisation that avoids executing the (now) following "if" block when edges == 0.  It's not needed after the (now) following "if" block, since edges is only update from adjustedEdges when adjustedEdges != 0.